pkg/operator/baremetal_pod.go

```diff
@@ -113,6 +144,44 @@ func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *Oper
 	return err
 }
 
+// we define this service in code instead of a yaml manifest because
+// we only want it to exist when metal3 is deployed
+func newMetal3Service(config *OperatorConfig, baremetalProvisioningConfig BaremetalProvisioningConfig) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "metal3-metrics",
+			Namespace: config.TargetNamespace,
+			Annotations: map[string]string{
+				maoOwnedAnnotation: "",
+				// copied from settings for machine-api-controllers
+				// service in
+				// 0000_30_machine-api-operator_10_service.yaml
+				"service.alpha.openshift.io/serving-cert-secret-name":    certStoreName,
+				"exclude.release.openshift.io/internal-openshift-hosted": "true",
+			},
+			Labels: map[string]string{
+				"k8s-app": "controller",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeNodePort,
+			Ports: []corev1.ServicePort{
+				{
+					Name: "metrics",
+					Port: metal3ExposeMetricsPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.String,
+						StrVal: "metrics",
+					},
+				},
+			},
+			Selector: map[string]string{
+				"k8s-app": "controller",
+			},
```

Comment:
Is this not going to select the Machine API core components as well? Is there a specific label that is applied to only the metal3 components?

Reply:
Same concern as above, this is going to try and expose the port on the `machine-api-onctrollers` pod as well isn't it? Or will this only come into effect when they're running in their own namespace?

If you check the endpoints for this service, which pods has it targeted?

Reply:
I see it includes the `machine-api-controllers` pod as well as the `metal3` pod. Those both run in the `openshift-machine-api` namespace.

Reply:
In that case, we need to verify that prometheus and in particular alerts are happy. IIRC there's an alert that first if any servicemonitor target is down, I would expect `machine-api-controllers` to show as down in this case since it won't have the correct port open?

Reply:
This can may be match a specific label that should be added/replaced in the baremetal deployment so only those endpoints are covered by the service.